csharp/src/Ice/IceSSL/TransceiverI.cs

```diff
@@ -393,34 +406,97 @@ private bool startAuthenticate(IceInternal.AsyncCallback callback, object state)
             {
                 if(!_incoming)
                 {
-                    //
                     // Client authentication.
-                    //
-                    _writeResult = _sslStream.AuthenticateAsClientAsync(_host,
-                                                                        _instance.certs(),
-                                                                        _instance.protocols(),
-                                                                        _instance.checkCRL() > 0);
-                    _writeResult.ContinueWith(task => callback(state), TaskScheduler.Default);
+                    if (_instance.initializationData().clientAuthenticationOptions
+                        is SslClientAuthenticationOptions clientAuthenticationOptions)
+                    {
+                        _writeResult = _sslStream.AuthenticateAsClientAsync(clientAuthenticationOptions);
+                        _writeResult.ContinueWith(
+                            task =>
+                            {
+                                try
+                                {
+                                    _verified = task.IsCompletedSuccessfully;
+                                    if (_verified)
+                                    {
+                                        _cipher = _sslStream.CipherAlgorithm.ToString();
+                                        if (_sslStream.RemoteCertificate is X509Certificate2 remoteCertificate)
+                                        {
+                                            _certs = [remoteCertificate];
```

Comment:
We can provide the complete certificate chain here. It is up to the remote certificate validation callback to build one. We provide the peer certificate that was validated.